src/plugins/guided_onboarding/public/services/api.ts

```diff
@@ -71,11 +74,12 @@ export class ApiService implements GuidedOnboardingApi {
    * Subsequently, the observable is updated automatically, when the state changes.
    */
   public fetchActiveGuideState$(): Observable<GuideState | undefined> {
-    return this.onboardingGuideState$.pipe(
-      concatMap((state) =>
-        !state && !this.isGuideStateLoading ? this.createGetStateObservable() : of(state)
-      )
-    );
+    const currentState = this.guideState$.value;
+    if (!currentState && !this.isGuideStateLoading && !this.isGuideStateInitialized) {
+      this.isGuideStateLoading = true;
```

Comment:
Can you explain why this is needed here?

Reply:
Sure, I'll add a comment to the code